File: gamslib/aircraft/aircraft-scalar.py
#  LP written by GAMS Convert at 12/13/18 10:24:29
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         57       28        0       29        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         73       73        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        222      222        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x5 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x10 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x11 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x12 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x17 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x18 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x19 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x20 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x21 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x22 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x23 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x24 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x25 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x26 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x27 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x28 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x29 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x30 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x31 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x32 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x33 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x34 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x35 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x36 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x37 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x38 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x39 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x40 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x41 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x42 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x43 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x44 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x45 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x46 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x47 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x48 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x49 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x50 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x51 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x52 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x53 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x54 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x55 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x56 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x57 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x58 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x59 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x60 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x61 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x62 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x63 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x64 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x65 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x66 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x67 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x68 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x69 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x70 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x71 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x72 = Var(within=Reals,bounds=(0,None),initialize=0)

m.obj = Objective(expr=   m.x71 + m.x72, sense=minimize)

m.c1 = Constraint(expr=   m.x1 + m.x2 + m.x3 + m.x4 + m.x5 <= 10)

m.c2 = Constraint(expr=   m.x6 + m.x7 + m.x8 + m.x9 + m.x10 <= 19)

m.c3 = Constraint(expr=   m.x11 + m.x12 + m.x13 + m.x14 + m.x15 <= 25)

m.c4 = Constraint(expr=   m.x16 + m.x17 + m.x18 + m.x19 + m.x20 <= 15)

m.c5 = Constraint(expr= - 16*m.x1 - 9*m.x16 + m.x21 <= 0)

m.c6 = Constraint(expr= - 16*m.x1 - 9*m.x16 + m.x22 <= 0)

m.c7 = Constraint(expr= - 16*m.x1 - 9*m.x16 + m.x23 <= 0)

m.c8 = Constraint(expr= - 16*m.x1 - 9*m.x16 + m.x24 <= 0)

m.c9 = Constraint(expr= - 16*m.x1 - 9*m.x16 + m.x25 <= 0)

m.c10 = Constraint(expr= - 15*m.x2 - 10*m.x7 - 5*m.x12 - 11*m.x17 + m.x26 <= 0)

m.c11 = Constraint(expr= - 15*m.x2 - 10*m.x7 - 5*m.x12 - 11*m.x17 + m.x27 <= 0)

m.c12 = Constraint(expr= - 15*m.x2 - 10*m.x7 - 5*m.x12 - 11*m.x17 + m.x28 <= 0)

m.c13 = Constraint(expr= - 15*m.x2 - 10*m.x7 - 5*m.x12 - 11*m.x17 + m.x29 <= 0)

m.c14 = Constraint(expr= - 15*m.x2 - 10*m.x7 - 5*m.x12 - 11*m.x17 + m.x30 <= 0)

m.c15 = Constraint(expr= - 28*m.x3 - 14*m.x8 - 22*m.x18 + m.x31 <= 0)

m.c16 = Constraint(expr= - 28*m.x3 - 14*m.x8 - 22*m.x18 + m.x32 <= 0)

m.c17 = Constraint(expr= - 28*m.x3 - 14*m.x8 - 22*m.x18 + m.x33 <= 0)

m.c18 = Constraint(expr= - 28*m.x3 - 14*m.x8 - 22*m.x18 + m.x34 <= 0)

m.c19 = Constraint(expr= - 28*m.x3 - 14*m.x8 - 22*m.x18 + m.x35 <= 0)

m.c20 = Constraint(expr= - 23*m.x4 - 15*m.x9 - 7*m.x14 - 17*m.x19 + m.x36 <= 0)

m.c21 = Constraint(expr= - 23*m.x4 - 15*m.x9 - 7*m.x14 - 17*m.x19 + m.x37 <= 0)

m.c22 = Constraint(expr= - 23*m.x4 - 15*m.x9 - 7*m.x14 - 17*m.x19 + m.x38 <= 0)

m.c23 = Constraint(expr= - 23*m.x4 - 15*m.x9 - 7*m.x14 - 17*m.x19 + m.x39 <= 0)

m.c24 = Constraint(expr= - 23*m.x4 - 15*m.x9 - 7*m.x14 - 17*m.x19 + m.x40 <= 0)

m.c25 = Constraint(expr= - 81*m.x5 - 57*m.x10 - 29*m.x15 - 55*m.x20 + m.x41 <= 0)

m.c26 = Constraint(expr= - 81*m.x5 - 57*m.x10 - 29*m.x15 - 55*m.x20 + m.x42 <= 0)

m.c27 = Constraint(expr= - 81*m.x5 - 57*m.x10 - 29*m.x15 - 55*m.x20 + m.x43 <= 0)

m.c28 = Constraint(expr= - 81*m.x5 - 57*m.x10 - 29*m.x15 - 55*m.x20 + m.x44 <= 0)

m.c29 = Constraint(expr= - 81*m.x5 - 57*m.x10 - 29*m.x15 - 55*m.x20 + m.x45 <= 0)

m.c30 = Constraint(expr=   m.x21 + m.x46 == 200)

m.c31 = Constraint(expr=   m.x22 + m.x47 == 220)

m.c32 = Constraint(expr=   m.x23 + m.x48 == 250)

m.c33 = Constraint(expr=   m.x24 + m.x49 == 270)

m.c34 = Constraint(expr=   m.x25 + m.x50 == 300)

m.c35 = Constraint(expr=   m.x26 + m.x51 == 50)

m.c36 = Constraint(expr=   m.x27 + m.x52 == 150)

m.c37 = Constraint(expr=   m.x28 + m.x53 == 0)

m.c38 = Constraint(expr=   m.x29 + m.x54 == 0)

m.c39 = Constraint(expr=   m.x30 + m.x55 == 0)

m.c40 = Constraint(expr=   m.x31 + m.x56 == 140)

m.c41 = Constraint(expr=   m.x32 + m.x57 == 160)

m.c42 = Constraint(expr=   m.x33 + m.x58 == 180)

m.c43 = Constraint(expr=   m.x34 + m.x59 == 200)

m.c44 = Constraint(expr=   m.x35 + m.x60 == 220)

m.c45 = Constraint(expr=   m.x36 + m.x61 == 10)

m.c46 = Constraint(expr=   m.x37 + m.x62 == 50)

m.c47 = Constraint(expr=   m.x38 + m.x63 == 80)

m.c48 = Constraint(expr=   m.x39 + m.x64 == 100)

m.c49 = Constraint(expr=   m.x40 + m.x65 == 340)

m.c50 = Constraint(expr=   m.x41 + m.x66 == 580)

m.c51 = Constraint(expr=   m.x42 + m.x67 == 600)

m.c52 = Constraint(expr=   m.x43 + m.x68 == 620)

m.c53 = Constraint(expr=   m.x44 + m.x69 == 0)

m.c54 = Constraint(expr=   m.x45 + m.x70 == 0)

m.c55 = Constraint(expr= - 18*m.x1 - 21*m.x2 - 18*m.x3 - 16*m.x4 - 10*m.x5 - 15*m.x7 - 16*m.x8 - 14*m.x9 - 9*m.x10
                         - 10*m.x12 - 9*m.x14 - 6*m.x15 - 17*m.x16 - 16*m.x17 - 17*m.x18 - 15*m.x19 - 10*m.x20 + m.x71
                         == 0)

m.c56 = Constraint(expr= - 2.6*m.x46 - 0.65*m.x47 - 4.55*m.x48 - 2.6*m.x49 - 2.6*m.x50 - 3.9*m.x51 - 9.1*m.x52
                         - 0.7*m.x56 - 1.4*m.x57 - 2.8*m.x58 - 1.4*m.x59 - 0.7*m.x60 - 1.4*m.x61 - 1.4*m.x62 - 2.1*m.x63
                         - 1.4*m.x64 - 0.7*m.x65 - 0.1*m.x66 - 0.8*m.x67 - 0.1*m.x68 + m.x72 == 0)

File: gamslib/aircraft/aircraft-scalar.jl
#  LP written by GAMS Convert at 10/11/20 12:15:56
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         57       28        0       29        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         73       73        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        222      222        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x5, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)
@variable(m, 0 <= x10, start=0)
@variable(m, 0 <= x11, start=0)
@variable(m, 0 <= x12, start=0)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= x17, start=0)
@variable(m, 0 <= x18, start=0)
@variable(m, 0 <= x19, start=0)
@variable(m, 0 <= x20, start=0)
@variable(m, 0 <= x21, start=0)
@variable(m, 0 <= x22, start=0)
@variable(m, 0 <= x23, start=0)
@variable(m, 0 <= x24, start=0)
@variable(m, 0 <= x25, start=0)
@variable(m, 0 <= x26, start=0)
@variable(m, 0 <= x27, start=0)
@variable(m, 0 <= x28, start=0)
@variable(m, 0 <= x29, start=0)
@variable(m, 0 <= x30, start=0)
@variable(m, 0 <= x31, start=0)
@variable(m, 0 <= x32, start=0)
@variable(m, 0 <= x33, start=0)
@variable(m, 0 <= x34, start=0)
@variable(m, 0 <= x35, start=0)
@variable(m, 0 <= x36, start=0)
@variable(m, 0 <= x37, start=0)
@variable(m, 0 <= x38, start=0)
@variable(m, 0 <= x39, start=0)
@variable(m, 0 <= x40, start=0)
@variable(m, 0 <= x41, start=0)
@variable(m, 0 <= x42, start=0)
@variable(m, 0 <= x43, start=0)
@variable(m, 0 <= x44, start=0)
@variable(m, 0 <= x45, start=0)
@variable(m, 0 <= x46, start=0)
@variable(m, 0 <= x47, start=0)
@variable(m, 0 <= x48, start=0)
@variable(m, 0 <= x49, start=0)
@variable(m, 0 <= x50, start=0)
@variable(m, 0 <= x51, start=0)
@variable(m, 0 <= x52, start=0)
@variable(m, 0 <= x53, start=0)
@variable(m, 0 <= x54, start=0)
@variable(m, 0 <= x55, start=0)
@variable(m, 0 <= x56, start=0)
@variable(m, 0 <= x57, start=0)
@variable(m, 0 <= x58, start=0)
@variable(m, 0 <= x59, start=0)
@variable(m, 0 <= x60, start=0)
@variable(m, 0 <= x61, start=0)
@variable(m, 0 <= x62, start=0)
@variable(m, 0 <= x63, start=0)
@variable(m, 0 <= x64, start=0)
@variable(m, 0 <= x65, start=0)
@variable(m, 0 <= x66, start=0)
@variable(m, 0 <= x67, start=0)
@variable(m, 0 <= x68, start=0)
@variable(m, 0 <= x69, start=0)
@variable(m, 0 <= x70, start=0)
@variable(m, 0 <= x71, start=0)
@variable(m, 0 <= x72, start=0)

@objective(m, Min, x71 + x72)

@constraint(m, x1 + x2 + x3 + x4 + x5 <= 10)

@constraint(m, x6 + x7 + x8 + x9 + x10 <= 19)

@constraint(m, x11 + x12 + x13 + x14 + x15 <= 25)

@constraint(m, x16 + x17 + x18 + x19 + x20 <= 15)

@constraint(m,  - 16*x1 - 9*x16 + x21 <= 0)

@constraint(m,  - 16*x1 - 9*x16 + x22 <= 0)

@constraint(m,  - 16*x1 - 9*x16 + x23 <= 0)

@constraint(m,  - 16*x1 - 9*x16 + x24 <= 0)

@constraint(m,  - 16*x1 - 9*x16 + x25 <= 0)

@constraint(m,  - 15*x2 - 10*x7 - 5*x12 - 11*x17 + x26 <= 0)

@constraint(m,  - 15*x2 - 10*x7 - 5*x12 - 11*x17 + x27 <= 0)

@constraint(m,  - 15*x2 - 10*x7 - 5*x12 - 11*x17 + x28 <= 0)

@constraint(m,  - 15*x2 - 10*x7 - 5*x12 - 11*x17 + x29 <= 0)

@constraint(m,  - 15*x2 - 10*x7 - 5*x12 - 11*x17 + x30 <= 0)

@constraint(m,  - 28*x3 - 14*x8 - 22*x18 + x31 <= 0)

@constraint(m,  - 28*x3 - 14*x8 - 22*x18 + x32 <= 0)

@constraint(m,  - 28*x3 - 14*x8 - 22*x18 + x33 <= 0)

@constraint(m,  - 28*x3 - 14*x8 - 22*x18 + x34 <= 0)

@constraint(m,  - 28*x3 - 14*x8 - 22*x18 + x35 <= 0)

@constraint(m,  - 23*x4 - 15*x9 - 7*x14 - 17*x19 + x36 <= 0)

@constraint(m,  - 23*x4 - 15*x9 - 7*x14 - 17*x19 + x37 <= 0)

@constraint(m,  - 23*x4 - 15*x9 - 7*x14 - 17*x19 + x38 <= 0)

@constraint(m,  - 23*x4 - 15*x9 - 7*x14 - 17*x19 + x39 <= 0)

@constraint(m,  - 23*x4 - 15*x9 - 7*x14 - 17*x19 + x40 <= 0)

@constraint(m,  - 81*x5 - 57*x10 - 29*x15 - 55*x20 + x41 <= 0)

@constraint(m,  - 81*x5 - 57*x10 - 29*x15 - 55*x20 + x42 <= 0)

@constraint(m,  - 81*x5 - 57*x10 - 29*x15 - 55*x20 + x43 <= 0)

@constraint(m,  - 81*x5 - 57*x10 - 29*x15 - 55*x20 + x44 <= 0)

@constraint(m,  - 81*x5 - 57*x10 - 29*x15 - 55*x20 + x45 <= 0)

@constraint(m, x21 + x46 == 200)

@constraint(m, x22 + x47 == 220)

@constraint(m, x23 + x48 == 250)

@constraint(m, x24 + x49 == 270)

@constraint(m, x25 + x50 == 300)

@constraint(m, x26 + x51 == 50)

@constraint(m, x27 + x52 == 150)

@constraint(m, x28 + x53 == 0)

@constraint(m, x29 + x54 == 0)

@constraint(m, x30 + x55 == 0)

@constraint(m, x31 + x56 == 140)

@constraint(m, x32 + x57 == 160)

@constraint(m, x33 + x58 == 180)

@constraint(m, x34 + x59 == 200)

@constraint(m, x35 + x60 == 220)

@constraint(m, x36 + x61 == 10)

@constraint(m, x37 + x62 == 50)

@constraint(m, x38 + x63 == 80)

@constraint(m, x39 + x64 == 100)

@constraint(m, x40 + x65 == 340)

@constraint(m, x41 + x66 == 580)

@constraint(m, x42 + x67 == 600)

@constraint(m, x43 + x68 == 620)

@constraint(m, x44 + x69 == 0)

@constraint(m, x45 + x70 == 0)

@constraint(m,  - 18*x1 - 21*x2 - 18*x3 - 16*x4 - 10*x5 - 15*x7 - 16*x8 - 14*x9 - 9*x10 - 10*x12 - 9*x14 - 6*x15
     - 17*x16 - 16*x17 - 17*x18 - 15*x19 - 10*x20 + x71 == 0)

@constraint(m,  - 2.6*x46 - 0.65*x47 - 4.55*x48 - 2.6*x49 - 2.6*x50 - 3.9*x51 - 9.1*x52 - 0.7*x56 - 1.4*x57 - 2.8*x58
     - 1.4*x59 - 0.7*x60 - 1.4*x61 - 1.4*x62 - 2.1*x63 - 1.4*x64 - 0.7*x65 - 0.1*x66 - 0.8*x67 - 0.1*x68 + x72 == 0)
